src/otx/v2/adapters/torch/mmengine/dataset.py

```diff
@@ -0,0 +1,309 @@
+        val_data_roots: str | None = None,
+        val_ann_files: str | None = None,
+        test_data_roots: str | None = None,
+        test_ann_files: str | None = None,
+        unlabeled_data_roots: str | None = None,
+        unlabeled_file_list: str | None = None,
+        data_format: str | None = None,
+    ) -> None:
+        r"""MMEngine's Dataset class.
+
+        Args:
+            task (Optional[Union[TaskType, str]], optional): The task type of the dataset want to load.
+                Defaults to None.
+            train_type (Optional[Union[TrainType, str]], optional): The train type of the dataset want to load.
+                Defaults to None.
+            train_data_roots (Optional[str], optional): The root address of the dataset to be used for training.
+                Defaults to None.
+            train_ann_files (Optional[str], optional): Location of the annotation file for the dataset
+                to be used for training. Defaults to None.
+            val_data_roots (Optional[str], optional): The root address of the dataset
+                to be used for validation. Defaults to None.
+            val_ann_files (Optional[str], optional): Location of the annotation file for the dataset
+                to be used for validation. Defaults to None.
+            test_data_roots (Optional[str], optional): The root address of the dataset
+                to be used for testing. Defaults to None.
+            test_ann_files (Optional[str], optional): Location of the annotation file for the dataset
+                to be used for testing. Defaults to None.
+            unlabeled_data_roots (Optional[str], optional): The root address of the unlabeled dataset
+                to be used for training. Defaults to None.
+            unlabeled_file_list (Optional[str], optional): The file where the list of unlabeled images is declared.
+                Defaults to None.
+            data_format (Optional[str], optional): The format of the dataset. Defaults to None.
+        """
+        super().__init__(
+            task,
+            train_type,
+            train_data_roots,
+            train_ann_files,
+            val_data_roots,
+            val_ann_files,
+            test_data_roots,
+            test_ann_files,
+            unlabeled_data_roots,
+            unlabeled_file_list,
+            data_format,
+        )
+        self.scope = "mmengine"
+        self.dataset_registry: Registry = MMEngineRegistry().get("dataset")
+
+    def _initialize(self) -> None:
+        self.set_datumaro_adapters()  # Set self.dataset_entity & self.label_schema
+        self.base_dataset = self._get_sub_task_dataset()
+        self.initialize = True
+
+    def _get_sub_task_dataset(self) -> TorchDataset:
+        """Returns a TorchDataset object containing the data for the current sub-task.
+
+        Returns:
+            TorchDataset: A TorchDataset object containing the data for the current sub-task.
+        """
+
```

Comment:
AFAIU, this would be used as an interface? could we add `raise NotImplementedError`?